tests/test_pytorch_sparse_dataloader_api.py

```diff
@@ -0,0 +1,249 @@
+                                dataset_shape_x[0], dataset_shape_x[1]
+                            ),
+                            batch_size=BATCH_SIZE,
+                        )
+                        ingest_in_tiledb(
+                            uri=tiledb_uri_y,
+                            data=np.random.randint(
+                                low=0, high=NUM_OF_CLASSES, size=dataset_shape_y
+                            ),
+                            batch_size=BATCH_SIZE,
+                        )
+
+                        with tiledb.open(tiledb_uri_x) as x, tiledb.open(
+                            tiledb_uri_y
+                        ) as y:
+
+                            tiledb_dataset = PyTorchTileDBSparseDataset(
+                                x_array=x, y_array=y, batch_size=BATCH_SIZE
+                            )
+
+                            self.assertIsInstance(
+                                tiledb_dataset, torch.utils.data.IterableDataset
+                            )
+
+                            train_loader = torch.utils.data.DataLoader(
+                                tiledb_dataset, batch_size=None, num_workers=workers
+                            )
+
+                            # Train network
+                            net = Net(shape=dataset_shape_x[1:])
+                            criterion = torch.nn.CrossEntropyLoss()
+                            optimizer = optim.Adam(
+                                net.parameters(),
+                                lr=0.001,
+                                betas=(0.9, 0.999),
+                                eps=1e-08,
+                            )
+
+                            for epoch in range(
+                                1
+                            ):  # loop over the dataset multiple times
+                                for inputs, labels in train_loader:
+                                    # zero the parameter gradients
+                                    optimizer.zero_grad()
+                                    # forward + backward + optimize
+                                    outputs = net(inputs)
+                                    loss = criterion(
+                                        outputs, labels.type(torch.LongTensor)
+                                    )
+                                    loss.backward()
+                                    optimizer.step()
+
+    def test_except_with_diff_number_of_x_y_sparse_rows(self):
+        with tempfile.TemporaryDirectory() as tiledb_uri_x, tempfile.TemporaryDirectory() as tiledb_uri_y:
+            # Add one extra row on X
+            dataset_shape_x = (ROWS + 1, INPUT_SHAPES[0])
+            dataset_shape_y = (ROWS, (NUM_OF_CLASSES,))
+
+            ingest_in_tiledb_sparse(
+                uri=tiledb_uri_x,
+                data=create_sparse_array_one_hot_2d(
+                    dataset_shape_x[0], dataset_shape_x[1]
```

Comment:
Shorter `*dataset_shape_x`. Likewise for `dataset_shape_y` and all `create_sparse_array_one_hot_2d` calls.